spec/workers/webhook_worker_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe WebhookWorker do
+  describe "perform" do
+    let(:report) { { some: "json" } }
+    let(:webhook_uri) { "http://webhooks-rule.org/webhook" }
+    let(:secret_key) { nil }
+
+    context "with a secret key" do
+      let(:secret_key) { "this is a secret key" }
+
+      # http://www.freeformatter.com/hmac-generator.html
+      let(:expected_signature) { "d44d805a16b90922edb48f14ff4e858a5d3e39bd" }
+
+      let!(:stubbed_request) do
+        stub_request(:post, webhook_uri)
+          .with(body: report.to_json,
+                headers: { "X-LinkCheckerApi-Signature": expected_signature })
+          .to_return(status: 200)
+      end
+
+      before do
+        subject.perform(report, webhook_uri, secret_key)
+      end
+
+      it "generates a valid signature" do
+        expect(stubbed_request).to have_been_requested
```

Comment:
Might be more idiomatic to have this be along the lines of `expect(a_request(:post, webhook_uri).with(headers: {...})).to have_been_made` and then just do a more generic stub in the before block?